sdk/ml/azure-ai-ml/azure/ai/ml/exceptions.py

```diff
@@ -214,17 +212,15 @@ def __init__(
         message: str,
         no_personal_data_message: str,
         *args,
-        target: ErrorTarget = ErrorTarget.UNKNOWN,
-        error_category: ErrorCategory = ErrorCategory.UNKNOWN,
+        target: Union[str, ErrorTarget] = ErrorTarget.UNKNOWN,
+        error_category: Union[str, ErrorTarget] = ErrorCategory.UNKNOWN,
```

Comment:
Again, these categories should match.

Reply:
mypy check is considered `ErrorTarget.UNKNOWN` as `str` type, not `ErrorTarget `type

Reply:
That's fine. I should've been clearer. I meant that `error_category: Union[str, ErrorTarget]` should be `error_category: Union[str, ErrorCategory]`. Fixing the bug in DeploymentException's type annotations should make this clear and then make the mypy errors go away.

Reply:
fixed